### PR TITLE
Add custom type for gzip_proxied

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -110,7 +110,7 @@
   $gzip_disable                                              = 'msie6',
   $gzip_min_length                                           = 20,
   $gzip_http_version                                         = 1.1,
-  $gzip_proxied                                              = 'off',
+  Nginx::GzipProxied $gzip_proxied                           = 'off',
```
Comment (arjenz) [inline manifests/init.pp:113]: I don't see that type in `/types`?

File: REFERENCE.md [hunk 2]
```diff
@@ -21,35 +21,36 @@
...
+* [`nginx::resource::map`](#nginx--resource--map): Create a new mapping entry for NGINX
+* [`nginx::resource::server`](#nginx--resource--server): Create a virtual host
+* [`nginx::resource::snippet`](#nginx--resource--snippet): Create a reusable config snippet that can be included by other resources
+* [`nginx::resource::streamhost`](#nginx--resource--streamhost): Create a virtual steamhost
+* [`nginx::resource::upstream`](#nginx--resource--upstream): Create a new upstream proxy entry for NGINX
+* [`nginx::resource::upstream::member`](#nginx--resource--upstream--member): Create an upstream member inside the upstream block.
 
 ### Data types
 
-* [`Nginx::DebugConnection`](#nginxdebugconnection)
```
Comment (arjenz) [inline REFERENCE.md:36]: Most of REFERENCE.md appears to be rewritten, replacing all anchor references, maybe put this in a separate commit as this seems unrelated to the new type?

Comment (bastelfreak) [reply REFERENCE.md:36]: done

**Review by arjenz — APPROVED** [review]
